iceoryx_hoofs/container/include/iox/uninitialized_array.hpp

```diff
@@ -15,29 +15,30 @@
 //
 // SPDX-License-Identifier: Apache-2.0
 
-#ifndef IOX_HOOFS_CONTAINERS_UNINITIALIZED_ARRAY_HPP
-#define IOX_HOOFS_CONTAINERS_UNINITIALIZED_ARRAY_HPP
+#ifndef IOX_HOOFS_CONTAINER_UNINITIALIZED_ARRAY_HPP
+#define IOX_HOOFS_CONTAINER_UNINITIALIZED_ARRAY_HPP
 
 #include "iceoryx_hoofs/iceoryx_hoofs_types.hpp"
 
 #include <cstdint>
 
 namespace iox
 {
-namespace containers
-{
 /// @brief struct used as policy parameter in UninitializedArray to wrap an array with all elements zeroed
 /// @tparam ElementType is the array type
 /// @tparam Capacity is the array size
 template <typename ElementType, uint64_t Capacity>
 struct ZeroedBuffer
 {
+    struct alignas(ElementType) element_t
```

Comment:
@FerdinandSpitzschnueffler Good, this allows using it with `char` as expected. 

This should prevent the optimization that we stumbled about. It was a `strlen` optimization in gcc 8.x (specifically 8.4) that could maybe happen also in other circumstances, as the compiler may rightfully assume that `strlen` of a `char[1]` must be 0 (due to zero-termination).

The effects of various optimization can be observed in https://godbolt.org/z/ooqd9M5T3

Reply:
Thanks for the example!